lib/Auth/Source/Radius.php

```diff
@@ -133,77 +140,47 @@ public function __construct(array $info, array $config)
      */
     protected function login(string $username, string $password): array
     {
-        $radius = radius_auth_open();
-        if (!is_resource($radius)) {
-            throw new Exception("Insufficient memory available to create handle.");
-        }
+        $radius = new RadiusClient();
+        $response = false;
 
         // Try to add all radius servers, trigger a failure if no one works
-        $success = false;
         foreach ($this->servers as $server) {
-            if (!isset($server['port'])) {
-                $server['port'] = 1812;
-            }
-            if (
-                !radius_add_server(
-                    $radius,
-                    $server['hostname'],
-                    $server['port'],
-                    $server['secret'],
-                    $this->timeout,
-                    $this->retries
-                )
-            ) {
-                Logger::info(
-                    "Could not add radius server: " . radius_strerror($radius)
-                );
-                continue;
-            }
-            $success = true;
-        }
-        if (!$success) {
-            throw new Exception('Error adding radius servers, no servers available');
-        }
+            $radius->setServer($server['hostname']);
+            $radius->setAuthenticationPort($server['port']);
+            $radius->setSecret($server['secret']);
+            $radius->setDebug($this->debug);
+            $radius->setTimeout($this->timeout);
 
-        if (!radius_create_request($radius, \RADIUS_ACCESS_REQUEST)) {
-            throw new Exception(
-                'Error creating radius request: ' . radius_strerror($radius)
-            );
-        }
+            $httpUtils = new Utils\HTTP();
+            $radius->setNasIpAddress($httpUtils->getSelfHost());
 
-        if ($this->realm === null) {
-            radius_put_attr($radius, \RADIUS_USER_NAME, $username);
-        } else {
-            radius_put_attr($radius, \RADIUS_USER_NAME, $username . '@' . $this->realm);
-        }
-        radius_put_attr($radius, \RADIUS_USER_PASSWORD, $password);
+            if ($this->nasIdentifier !== null) {
+                $radius->setAttribute(self::RADIUS_NAS_IDENTIFIER, $this->nasIdentifier);
+            }
 
-        if ($this->nasIdentifier !== null) {
-            radius_put_attr($radius, \RADIUS_NAS_IDENTIFIER, $this->nasIdentifier);
-        }
+            if ($this->realm !== null) {
+                $radius->setRadiusSuffix($this->realm);
+            }
+            $response = $radius->accessRequest($username, $password);
 
-        $res = radius_send_request($radius);
-        if ($res !== \RADIUS_ACCESS_ACCEPT) {
-            switch ($res) {
-                case \RADIUS_ACCESS_REJECT:
-                    // Invalid username or password
-                    throw new \SimpleSAML\Error\Error('WRONGUSERPASS');
```

Comment:
This case is now missing. If you supply a wrong password as a user you do not get the correct page, but get an "Unhandled exception"

Reply:
Fixed this